src/NUnitFramework/tests/Constraints/EndsWithConstraintTests.cs

```diff
@@ -36,6 +38,102 @@ public void RespectsCulture()
+                constraint = new NotConstraint(constraint);
+
+            Assert.That(actual, constraint);
+        }
+
+        [TestCase(" ss", "ß")]
+        [TestCase(" ß", "ß")]
+        [TestCase(" ss", "s")]
+        [TestCase(" SS", "s")]
+        public void SpecifyCultureInfo(string actual, string expected)
+        {
+            var cultureInfo = new CultureInfo("de-DE");
+            // Get platform-specific StringComparison behavior
+            var shouldSucceed = actual.EndsWith(expected, false, cultureInfo);
+
+            Constraint constraint = Does.EndWith(expected).Using(cultureInfo);
+            if (!shouldSucceed)
+                constraint = new NotConstraint(constraint);
+
+            Assert.That(actual, constraint);
+        }
+
+        [Test]
+        public void UseDifferentComparisonTypes_ThrowsException()
+        {
+            var endsWithConstraint = (EndsWithConstraint)TheConstraint;
+
+            Assert.Multiple(() =>
+            {
+                // Invoke Using method before IgnoreCase
+                Assert.That(() => endsWithConstraint.Using(StringComparison.CurrentCulture).IgnoreCase,
+                    Throws.TypeOf<InvalidOperationException>());
+                Assert.That(() => endsWithConstraint.Using(StringComparison.InvariantCulture).IgnoreCase,
+                    Throws.TypeOf<InvalidOperationException>());
+                Assert.That(() => endsWithConstraint.Using(StringComparison.InvariantCultureIgnoreCase).IgnoreCase,
+                    Throws.TypeOf<InvalidOperationException>());
+                Assert.That(() => endsWithConstraint.Using(StringComparison.Ordinal).IgnoreCase,
+                    Throws.TypeOf<InvalidOperationException>());
+                Assert.That(() => endsWithConstraint.Using(StringComparison.OrdinalIgnoreCase).IgnoreCase,
+                    Throws.TypeOf<InvalidOperationException>());
+
+                // Invoke IgnoreCase before Using method
+                Assert.That(() => ((SubstringConstraint)endsWithConstraint.IgnoreCase).Using(StringComparison.CurrentCulture),
+                    Throws.TypeOf<InvalidOperationException>());
+                Assert.That(() => ((SubstringConstraint)endsWithConstraint.IgnoreCase).Using(StringComparison.InvariantCulture),
+                    Throws.TypeOf<InvalidOperationException>());
+                Assert.That(() => ((SubstringConstraint)endsWithConstraint.IgnoreCase).Using(StringComparison.InvariantCultureIgnoreCase),
+                    Throws.TypeOf<InvalidOperationException>());
+                Assert.That(() => ((SubstringConstraint)endsWithConstraint.IgnoreCase).Using(StringComparison.Ordinal).IgnoreCase,
+                    Throws.TypeOf<InvalidOperationException>());
+                Assert.That(() => ((SubstringConstraint)endsWithConstraint.IgnoreCase).Using(StringComparison.OrdinalIgnoreCase).IgnoreCase,
+                    Throws.TypeOf<InvalidOperationException>());
+            });
```

Comment:
The casts are now unnecessary as the implementation is on the base class.
Again not sure if these should throw or only modify the `StringComparison`?